source/testing/testing-components.md

```diff
@@ -364,29 +419,35 @@ const locationStub = Service.extend({
   }
 });
 
-moduleForComponent('location-indicator', 'Integration | Component | location indicator', {
-  integration: true,
+module('Integration | Component | location indicator', function(hooks) {
+  setupRenderingTest(hooks);
 
-  beforeEach: function () {
-    this.register('service:location-service', locationStub);
+  hooks.beforeEach(function(assert) {
+    this.owner.register('service:location-service', locationStub);
     // Calling inject puts the service instance in the context of the test,
     // making it accessible as "locationService" within each test
     this.inject.service('location-service', { as: 'locationService' });
-  }
-});
+  });
 
-test('should reveal current location', function(assert) {
-  this.render(hbs`{{location-indicator}}`);
-  assert.equal(this.$().text().trim(), 'You currently are located in New York, USA');
-});
+  test('should reveal current location', function(assert) {
+    await render(hbs`{{location-indicator}}`);
+    assert.equal(this.element.textContent.trim(),
+     'You currently are located in New York, USA');
+  });
+
+  test('should change displayed location when current location changes', async function (assert) {
+    await render(hbs`{{location-indicator}}`);
+
+    assert.equal(this.element.textContent.trim(),
+     'You currently are located in New York, USA', 'origin location should display');
+
+    this.set('locationService.city', 'Beijing');
+    this.set('locationService.country', 'China');
+    this.set('locationService.currentLocation', { x: 11111, y: 222222 });
```

Comment:
changing `currentLocation` seems pointless since the property is not actually used anywhere 🤔